### PR TITLE
BD-2204. Multicast servers. Changed fiosdk use array of api urls.

### PR DESCRIPTION
**Pull request by andreyvEze**

### CODE REVIEW

**Review by orkhan-huseyn — APPROVED** [review]

File: src/utils/utils.ts [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+  return array
+}
+
+export async function asyncWaterfall(
+  asyncFuncs: Array<Function>,
+  timeoutMs: number = 5000
+): Promise<any> {
+  let pending = asyncFuncs.length
+  const promises: Array<Promise<any>> = []
```
Comment (trukhilio) [inline src/utils/utils.ts:31]: If we know return params let's set type.

File: src/utils/utils.ts [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    array[currentIndex] = array[randomIndex]
+    array[randomIndex] = temporaryValue
+  }
+
+  return array
+}
+
+export async function asyncWaterfall(
+  asyncFuncs: Array<Function>,
+  timeoutMs: number = 5000
```
Comment (trukhilio) [inline src/utils/utils.ts]: Move timeout value to constant

File: src/utils/utils.ts [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    }
+    try {
+      const result = await Promise.race(promises)
+      if (result === 'async_waterfall_timed_out') {
+        promises.pop()
+        --pending
+      } else {
+        return result
+      }
+    } catch (error: any) {
```
Comment (trukhilio) [inline src/utils/utils.ts:57]: Is there any chance to set type here?

**Review by trukhilio — APPROVED** [review]
Check comments

**Review by ericbutz — APPROVED** [review]

**Pull request merged**